src/datastore/kvs/index.ts

```diff
@@ -105,6 +105,12 @@ export interface KeyValueStore<K extends Key, V extends Value> {
         return e;
     }
 
+    /** Returns an entry from the KVS, but only if it already
+     * exists in the store. */
+    getIfExists(key: K): Entry<K, V | null> | undefined {
+        return this._entries.get(key);
+    }
```

Comment:
I think<sup>[benchmark needed]</sup> this will actually end up being slower than just using `.get()` directly. The reason `.get()` always returns an Entry, even if it's empty, is to take advantage of Vue reactivity--if an icon shows up later (e.g. due to background fetching from the on-disk cache), the Entry will be updated directly, and Vue will be able to notice. Will explain further below.

Reply:
> I think[benchmark needed] this will actually end up being slower than just using `.get()` directly.

The logic for doing it that way actually wasn't optimization (well, time optimization anyway.) My reasoning here is these are just tabs, not bookmarks. They may have been opened from someone clicking a random link inside a page. Do we want to start creating `FaviconEntry`s in the store that aren't necessarily related to a stash item? It seems like that could be a memory leak unless there's something to prune those entries (I looked and didn't see anything that obviously looked like it was reaping them.)

If not, then synthesizing a `FaviconEntry` may result in `getIfExists` being called less times, but it's also possible that the next time it was called that there actually would be something.

Reply:
You're right, these don't get pruned, and they'll accumulate over time. And I think you're also right that we'll accumulate a lot more than we otherwise might because of normal page navigation.

Unfortunately, there's not an effective way to prune them without resorting to manual memory management, because JavaScript, `<sarcasm>`in its infinite wisdom`</sarcasm>`, does not have anything usefully resembling a weak map that lets you look up by key. This is apparently intentional, since it might [introduce non-determinism](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap).

So I think you're actually right that `getIfExists()` is probably a better approach for now, and I'll have to give some more thought to how favicons can be cleaned up. (One hack I thought of just now is to keep a generation number/dirty bit/something in a Vue ref that can be changed whenever entries are purged from the map, forcing Vue to re-compute the favicon properties and thus re-fetch any empty entries that are still in use. But I'll have to play with it a bit, I think.)

In any case, I don't think this PR needs to be held up on memory management; this can be cleaned up later.